lib/suites/Ed25519Signature2018.js

```diff
@@ -73,6 +75,19 @@ module.exports = class Ed25519Signature2018 extends LinkedDataSignature {
 
     // create detached content signature
     return encodedHeader + '..' + encodedSignature;
+
+    function _createSigningString() {
+
+      // FIXME: we need a Buffer implementation that works for browser
```

Comment:
Yes, Webpack provides a buffer polyfill (though from what I understand, @dlongley doesn't like it? not quite sure why.)

Reply:
In general, we should be writing code for the browser and polyfilling node, not the other way around. This is especially true when some core feature is currently never expected to be implemented in the browser. One property of polyfills is that they are intended to be temporary; they'll eventually go away, removing the need for sending them to the browser. In this light, the `Buffer` "polyfill" really isn't a polyfill. The `Buffer` "polyfill" is a foreverfill.

That being said, I'm not opposed to compromise in the interest of time. But we shouldn't be treating this sort of problem the same as, for example, using async/await or Promises and having to polyfill those for some browsers. This isn't write once and wait for browsers to catch up and we get to remove the overhead. This overhead will stick around until we manually remove it -- and that sort of overhead tends to pile up and overwhelm you before you know it.

Reply:
@dlongley another compromise here would be to pass the `signingString` to the handler in hex or base64, and let the handler implement a buffer if required in whatever way it chooses.  (e.g. forge)

Reply:
Also, looking at this, why do we need to use a Buffer here at all? Or, alternatively, check `verifyData.encoding` and use a Buffer when it makes sense, otherwise just do string concatenation for a `binary` string encoding -- which is probably what the browser version is doing.

Reply:
When we discussed this API the other day we said that it would accept a string or buffer.  I'm fine with always passing in a binary encoded string.  That *is* what you are suggesting?

Reply:
We've decided to use unit8array.